protocol/builder/produce_response_builder.go

```diff
@@ -0,0 +1,95 @@
+package builder
+
+import (
+	"sort"
+
+	kafkaapi "github.com/codecrafters-io/kafka-tester/protocol/api"
+	"github.com/codecrafters-io/kafka-tester/protocol/api/headers"
+)
+
+type ProduceResponseBuilder struct {
+	// topicName -> partitionIndex -> partitionResponse
+	topics         map[string]map[int32]kafkaapi.ProducePartitionResponse
+	throttleTimeMs int32
+}
+
+func NewProduceResponseBuilder() *ProduceResponseBuilder {
+	return &ProduceResponseBuilder{
+		topics:         make(map[string]map[int32]kafkaapi.ProducePartitionResponse),
+		throttleTimeMs: 0,
+	}
+}
+
+func (b *ProduceResponseBuilder) WithThrottleTimeMs(throttleTimeMs int32) *ProduceResponseBuilder {
+	b.throttleTimeMs = throttleTimeMs
+	return b
+}
+
+func (b *ProduceResponseBuilder) AddPartitionResponse(topicName string, partitionIndex int32, partitionResponse kafkaapi.ProducePartitionResponse) *ProduceResponseBuilder {
```

Comment:
Think hard about what one would actually need to customize, see if we can avoid split between `AddPartitionResponse` and `CreateAndAddErrorPartitionResponse`. Ideally we'd avoid a user needing to use `ProductionPartitionResponseBuilder` unless it saves significant complexity